conan/tools/premake/premakedeps.py

```diff
@@ -37,74 +109,217 @@ def __init__(self, deps_cpp_info):
+                   + list(build_req.items())
+
+        # Process dependencies and accumulate globally required data
+        pkg_files = []
+        dep_names = []
+        for require, dep in full_req:
+            dep_name = require.ref.name
+            dep_names.append(dep_name)
+
+            # Convert and aggregate dependency's
+            dep_cppinfo = dep.cpp_info.copy()
+            dep_cppinfo.set_relative_base_folder(dep.package_folder)
+            dep_aggregate = dep_cppinfo.aggregated_components()
+            
+            # Generate config dependent package variable and setup premake file
+            var_filename = PREMAKE_VAR_FILE.format(pkgname=dep_name, config=conf_suffix)
+            conf_filename = PREMAKE_CONF_FILE.format(pkgname=dep_name, config=conf_suffix)
+            self._output_lua_file(var_filename, [
+                PREMAKE_TEMPLATE_VAR.format(pkgname=dep_name, 
+                    config=conf_suffix, deps=_PremakeTemplate(dep_aggregate))
+            ])
+            self._output_lua_file(conf_filename, [
+                PREMAKE_TEMPLATE_CONF.format(
+                    pkgname=dep_name, config=conf_suffix, premake_varfile=var_filename,
+                    conf_consume_build=self._indent_string(
+                        PREMAKE_TEMPLATE_CONF_BUILD.format(pkgname=dep_name, config=conf_suffix)
+                    ),
+                    conf_consume_link=self._indent_string(
+                        PREMAKE_TEMPLATE_CONF_LINK.format(pkgname=dep_name, config=conf_suffix)
+                    )
+                )
+            ])
+
+            # Create list of all available profiles by searching on disk
+            file_pattern = PREMAKE_VAR_FILE.format(pkgname=dep_name, config="_*")
+            file_regex = PREMAKE_VAR_FILE.format(pkgname=re.escape(dep_name), config="_(([^_]*)_(.*))")
+            available_files = glob.glob(file_pattern)
+            # Add filename of current generations var file if not already present
+            if var_filename not in available_files:
+                available_files.append(var_filename)
+            profiles = [
+                (regex_res[0], regex_res.group(1), regex_res.group(2), regex_res.group(3)) for regex_res in [
+                    re.search(file_regex, file_name) for file_name in available_files
+                ]
+            ]
+            configurations = [profile[2] for profile in profiles]
+            architectures = list(dict.fromkeys([profile[3] for profile in profiles]))
+
+            # Fallback configuration (when user defined config is unknown -> prefer release or last)
+            fallback_configuration = "release"
+            if "release" not in configurations and len(configurations) > 0: 
+                fallback_configuration = configurations[-1]
+            
+            # Emit package premake file
+            pkg_files.append(PREMAKE_PKG_FILE.format(pkgname=dep_name))
+            self._output_lua_file(pkg_files[-1], [
```

Comment:
```suggestion
            pkg_filename = PREMAKE_PKG_FILE.format(pkgname=dep_name)
            pkg_files.append(pkg_filename)
            self._output_lua_file(pkg_filename, [
```

Just a small detail, we typically prefer this more explicit linear style